lib/ansible/modules/remote_management/manageiq/manageiq_tags.py

```diff
@@ -0,0 +1,258 @@
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+def manageiq_entities():
+    return {
+        'provider': 'providers', 'host': 'hosts', 'vm': 'vms',
+        'category': 'categories', 'cluster': 'clusters', 'data store': 'data_stores',
+        'group': 'groups', 'resource pool': 'resource_pools', 'service': 'services',
+        'service template': 'service_templates', 'template': 'templates',
+        'tenant': 'tenants', 'user': 'users', 'blueprint': 'blueprints'
+    }
+
+
+def query_resource_id(manageiq, resource_type, resource_name):
+    """ Query the resource name in ManageIQ.
+
+    Returns:
+        the resource id if it exists in manageiq, Fail otherwise.
+    """
+    resource = manageiq.find_collection_resource_by(resource_type, name=resource_name)
+    if resource:
+        return resource["id"]
+    else:
+        msg = "{resource_name} {resource_type} does not exist in manageiq".format(
+            resource_name=resource_name, resource_type=resource_type)
+        manageiq.module.fail_json(msg=msg)
+
+
+class ManageIQTags(object):
+    """
+        Object to execute tags management operations of manageiq resources.
+    """
+
+    def __init__(self, manageiq, resource_type, resource_id):
+        self.manageiq = manageiq
+
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+
+        self.resource_type = resource_type
+        self.resource_id = resource_id
+        self.resource_url = self.calc_resource_url()
+
+    def calc_resource_url(self):
+        """ Return the resource api url
+        """
+        return '{api_url}/{resource_type}/{resource_id}'.format(
+            api_url=self.api_url,
+            resource_type=self.resource_type,
+            resource_id=self.resource_id)
+
+    def full_tag_name(self, tag):
+        """ Returns the full tag name in manageiq
+        """
+        return '/managed/{tag_category}/{tag_name}'.format(
+            tag_category=tag['category'],
+            tag_name=tag['name'])
+
+    def query_resource_tags(self):
+        """ Returns a set of the full tag names assigned to the resource
+        """
+        url = '{resource_url}/tags?expand=resources'.format(resource_url=self.resource_url)
+        try:
+            response = self.client.get(url)
+        except Exception as e:
+            msg = "Failed to query {resource_type} tags: {error}".format(
+                resource_type=self.resource_type,
+                error=e)
+            self.module.fail_json(msg=msg)
+
+        tags = response.get('resources', [])
+        tags_set = set([tag['name'] for tag in tags])
+        return tags_set
+
+    def post_tags(self, tags, action):
+        """ Post the tags for the resource using ManageIQ rest api
```

Comment:
"Post the tags" is not very clear (and the fact it uses POST not interesting).  "Perform assign/unassign action" ?

More importantly I'd say a few words about what `tags` is.